packages/ui/src/components/SearchInput/Key.tsx

```diff
@@ -22,18 +23,43 @@ const Container = styled.div`
   }
 `
 
+export const KEYS_MATCH = {
+  Enter: '↵',
+  ' ': '␣',
+  Meta: '⌘',
+  Control: 'Ctrl',
+  Alt: 'Alt',
+  ArrowUp: '↑',
+  ArrowDown: '↓',
+  ArrowLeft: '←',
+  ArrowRight: '→',
+  Backspace: '⌫',
+  Delete: '⌦',
+  Escape: 'Esc',
+  CapsLock: 'Caps',
+} as const
+
 type KeyProps = {
-  children: string
+  children: KeyboardEvent['key']
   disabled?: boolean
 }
 
-export const Key = ({ children, disabled }: KeyProps) => (
-  <Container
-    data-disabled={disabled}
-    data-children-length={children.length > 1}
-  >
-    <Text as="span" variant="caption" sentiment="neutral" disabled={disabled}>
-      {children}
-    </Text>
-  </Container>
-)
+export const Key = ({ children, disabled }: KeyProps) => {
+  const isSpecialKey = useMemo(
+    () => Object.keys(KEYS_MATCH).find(key => key === children),
+    [children],
+  )
+
+  return (
+    <Container
+      data-disabled={disabled}
+      data-children-length={children.length > 1}
+    >
+      <Text as="span" variant="caption" sentiment="neutral" disabled={disabled}>
+        {isSpecialKey
+          ? KEYS_MATCH[children as keyof typeof KEYS_MATCH]
```

Comment:
do we need to cast here ? can't we have a typeguard function ?

Reply:
isSpecialKey is not well typed, that why you have to cast.

```
 const isSpecialKey = useMemo(
    () =>
      Object.keys(KEYS_MATCH).find(
        key => key === children,
      ) as (typeof KEYS_MATCH)[keyof typeof KEYS_MATCH],
    [children],
  )
```

Reply:
I did that too first but it's casting also isSpecialKey even though I agree it's cleaner this way